framework/web/User.php

```diff
@@ -437,17 +426,17 @@ public function loginRequired($checkAjax = true)
      * If you override this method, make sure you call the parent implementation
      * so that the event is triggered.
      * @param IdentityInterface $identity the user identity information
-     * @param boolean $cookieBased whether the login is cookie-based
      * @param integer $duration number of seconds that the user can remain in logged-in status.
      * If 0, it means login till the user closes the browser or the session is manually destroyed.
+     * @param boolean $cookieBased whether the login is cookie-based
      * @return boolean whether the user should continue to be logged in
      */
-    protected function beforeLogin($identity, $cookieBased, $duration)
+    protected function beforeLogin($identity, $duration, $cookieBased)
```

Comment:
You can't change signature since it's part of API already and it will break backwards compatibility.